File: src/test/java/ValidParenthesesTest.java
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class ValidParenthesesTest {
    @ParameterizedTest
    @ValueSource(strings = { "()", "()[]{}", "{[()]}", "{((()[]{}[[()(){{{[]}}}]]))}" })
    void should_accept_parentheses(String sequence) {
        ValidParentheses solution = new ValidParentheses();
        assertTrue(solution.isValid(sequence));
    }

    @ParameterizedTest
    @ValueSource(strings = { "())", "(){[}]", "()]}", "(((]]]{{{}}}" })
    void should_not_accept_parentheses(String sequence) {
        ValidParentheses solution = new ValidParentheses();
        assertFalse(solution.isValid(sequence));
    }
}

File: src/test/kotlin/ValidParenthesesTest.kt
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource

internal class ValidParenthesesTest {
    @ParameterizedTest
    @ValueSource(strings = ["()", "()[]{}", "{[()]}", "{((()[]{}[[()(){{{[]}}}]]))}"])
    fun should_accept_parentheses(sequence: String?) {
        val solution = ValidParentheses()
        Assertions.assertTrue(solution.isValid(sequence!!))
    }

    @ParameterizedTest
    @ValueSource(strings = ["())", "(){[}]", "()]}", "(((]]]{{{}}}"])
    fun should_not_accept_parentheses(sequence: String?) {
        val solution = ValidParentheses()
        Assertions.assertFalse(solution.isValid(sequence!!))
    }
}
